scripts/app.js

```diff
@@ -123,16 +127,22 @@ buttonZoomReset.addEventListener("click", function(event) {
 });
 
 function zoom(zoomValue) {
-    const currentFontSize = parseInt(window.getComputedStyle(input).fontSize);
+    const fontSizeRem = parseFloat(window.getComputedStyle(input).fontSize) / parseFloat(window.getComputedStyle(document.documentElement).fontSize);
+    const lineHeightRem = parseFloat(window.getComputedStyle(input).lineHeight) / parseFloat(window.getComputedStyle(document.documentElement).fontSize);
+    const outputLineHeightRem = parseFloat(window.getComputedStyle(output).lineHeight) / parseFloat(window.getComputedStyle(document.documentElement).fontSize);
 
-    if (currentFontSize + zoomValue < 10 || currentFontSize + zoomValue > 40) {
+    const newFontSizeRem = fontSizeRem + zoomValue;
+    const newLineHeightRem = lineHeightRem + zoomValue;
+    const newOutputLineHeightRem = outputLineHeightRem + zoomValue;
+
+    if (newFontSizeRem < 0.5 || newFontSizeRem > 2) {
```

Comment:
This is really minor, but you can move this condition before the initialization of all constants: currently, you are initializing quite a few constants, which might never be used.

```
const fontSizeRem = parseFloat(window.getComputedStyle(input).fontSize) / documentFontSize;
const newFontSizeRem = fontSizeRem + zoomValue;

if (newFontSizeRem < 0.5 || newFontSizeRem > 2) {
    return;
}

const lineHeightRem = parseFloat(window.getComputedStyle(input).lineHeight) / documentFontSize;
const outputLineHeightRem = parseFloat(window.getComputedStyle(output).lineHeight) / documentFontSize;

const newLineHeightRem = lineHeightRem + zoomValue;
const newOutputLineHeightRem = outputLineHeightRem + zoomValue;

...
```